qiskit_nature/problems/sampling/protein_folding/bead_distances/distance_map.py

```diff
@@ -0,0 +1,150 @@
+
+    @property
+    def peptide(self) -> Peptide:
+        """Returns a peptide."""
+        return self._peptide
+
+    @property
+    def distance_map(self) -> DefaultDict[BaseBead, Dict[BaseBead, OperatorBase]]:
+        """Returns a distance map."""
+        return self._distance_map
+
+    @property
+    def num_distances(self) -> int:
+        """Returns the number of distances calculated."""
+        return self._num_distances
+
+    def _first_neighbor(
+        self,
+        peptide: Peptide,
+        lower_bead_ind: int,
+        is_side_chain_upper: int,
+        upper_bead_ind: int,
+        is_side_chain_lower: int,
+        lambda_1: float,
+        pair_energies: np.ndarray,
+        pair_energies_multiplier: float = 0.1,
+    ) -> Union[PauliSumOp, PauliOp]:
+        """
+        Creates first nearest neighbor interaction if beads are in contact
+        and at a distance of 1 unit from each other. Otherwise, a large positive
+        energetic penalty is added. Here, the penalty depends on the neighboring
+        beads of interest (i and j), that is, :math:`lambda_0 > 6*(j -i + 1)*lambda_1 + e_{ij}`.
+        Here, we chose, :math:`lambda_0 = 7*(j- 1 + 1)`.
+
+        Args:
+            peptide: A Peptide object that includes all information about a protein.
+            lower_bead_ind: Backbone bead at turn i.
+            upper_bead_ind: Backbone bead at turn j (j > i).
+            is_side_chain_upper: Side chain on backbone bead j.
+            is_side_chain_lower: Side chain on backbone bead i.
+            lambda_1: Constraint to penalize local overlap between
+                     beads within a nearest neighbor contact.
+            pair_energies: Numpy array of pair energies for amino acids.
+            pair_energies_multiplier: A constant that multiplies pair energy contributions.
+
+        Returns:
+            expression: Contribution to an energetic Hamiltonian.
+        """
+        bounding_constant = 7
+        lambda_0 = bounding_constant * (upper_bead_ind - lower_bead_ind + 1) * lambda_1
+        lower_bead = peptide.get_main_chain[lower_bead_ind - 1]
+        upper_bead = peptide.get_main_chain[upper_bead_ind - 1]
+        if is_side_chain_upper == 1:
+            lower_bead = lower_bead.side_chain[0]
+        if is_side_chain_lower == 1:
+            upper_bead = upper_bead.side_chain[0]
+        energy = pair_energies[lower_bead_ind][is_side_chain_upper][upper_bead_ind][
+            is_side_chain_lower
+        ]
+        x = self.distance_map[lower_bead][upper_bead]
+        expression = lambda_0 * (
+            x - _build_full_identity(x.num_qubits)
+        ) + pair_energies_multiplier * energy * _build_full_identity(x.num_qubits)
+        return _fix_qubits(expression)
+
+    def _second_neighbor(
+        self,
+        peptide: Peptide,
+        lower_bead_ind: int,
+        is_side_chain_lower: int,
+        upper_bead_ind: int,
+        is_side_chain_upper: int,
```

Comment:
FYI: this is probably the other place where I commented on the order previously.. Here, the order in the docstring also does not match the actual order.

@manoelmarques Isn't the positional argument order in the docstring something which pylint should catch? If not, could we look into a tool which does catch this? `pydocstyle` might be a candidate but I would have to double check that..